### ISSUE TITLE
It takes long time to train 

### ISSUE THREAD

**Issue by juntawu** [saic-vul/ritm_interactive_segmentation#5]
Hello. Thanks for you work.

**Comment by haoyuying** [comment]
@juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。

**Comment by liyuxuan89** [comment]
I trained hrnet18s on one 1080ti for 200 epochs. It took approximate 20mins per epoch. The result is lower than reported. I wonder if this is normal. 
![企业微信截图_20210623172604](https://user-images.githubusercontent.com/43963025/123074253-b89e1100-d449-11eb-925f-b3b324420ae1.png)

**Comment by qinliuliuqin** [comment]
> > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > 
> > 
> > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> 
> Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。

训练结束后会做validation，你可以去过一遍代码，这个代码写得很好。validation的时候会停顿下，但不会很久，而且会有进度条显示。

**Comment by ty199931** [comment]
> > > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > > 
> > > 
> > > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> > 
> > 
> > Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。
> 
> 训练结束后会做validation，你可以去过一遍代码，这个代码写得很好。validation的时候会停顿下，但不会很久，而且会有进度条显示。

我看了代码了，然后也挨个代码打断点找问题，发现他有的时候连for循环都进不去，如果你们都没问题的话，那可能是我的电脑的原因？    或者我的数据集有问题？

**Comment by hyalvin** [comment]
> > > > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > > > 
> > > > 
> > > > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> > > 
> > > 
> > > Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。
> > 
> > 
> > 训练结束后会做validation，你可以去过一遍代码，这个代码写得很好。validation的时候会停顿下，但不会很久，而且会有进度条显示。
> 
> 我看了代码了，然后也挨个代码打断点找问题，发现他有的时候连for循环都进不去，如果你们都没问题的话，那可能是我的电脑的原因？ 或者我的数据集有问题？

请问你们有每个epoch训练loss都重新开始的问题么，感觉每个epoch都是独立的

**Comment by hyalvin** [comment]
> I trained hrnet18s on one 1080ti for 200 epochs. It took approximate 20mins per epoch. The result is lower than reported. I wonder if this is normal. ![企业微信截图_20210623172604](https://user-images.githubusercontent.com/43963025/123074253-b89e1100-d449-11eb-925f-b3b324420ae1.png)

hello, may i ask how you get this results? My validation process only gives me the validation loss result.

**Comment by xiangyunfan** [comment]
> > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > 
> > 
> > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> 
> Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。

请问后来这个问题是怎么解决的呢，我训练自己的数据集也遇到了同样的问题，训练完第一个epoch到validation就卡死了

**Comment by yangshunDragon** [comment]
> > > > 久是吗？我也不敢去乱点。
> > > 
> > > 
> > > 训练结束后会做validation，你可以去过一遍代码，这个代码写得很好。validation的时候会停顿下，但不会很久，而且会有进度条显示。
> > 
> > 
> > 我看了代码了，然后也挨个代码打断点找问题，发现他有的时候连for循环都进不去，如果你们都没问题的话，那可能是我的电脑的原因？ 或者我的数据集有问题？
> 
> 请问你们有每个epoch训练loss都重新开始的问题么，感觉每个epoch都是独立的

请问这个问题后来你怎么解决的呢？

**Comment by yangshunDragon** [comment]
> > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > 
> > 
> > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> 
> Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。

请问这个问题后来你怎么解决的呢？

**Comment by yangshunDragon** [comment]
> > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > 
> > 
> > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> 
> Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。

请问这个问题后来你怎么解决的呢？

**Comment by yangshunDragon** [comment]
> > > > @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > > 
> > > 
> > > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> > 
> > 
> > Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。
> 
> 请问后来这个问题是怎么解决的呢，我训练自己的数据集也遇到了同样的问题，训练完第一个epoch到validation就卡死了

请问这个问题后来你怎么解决的呢？

**Comment by ty199931** [comment]
好像是因为影像的原因，把没有标签的剔除掉就可以了




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年5月7日(星期六) 下午5:42
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [saic-vul/ritm_interactive_segmentation] It takes long time to train  (#5)





     
@juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
  
The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
  
Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。
  
请问后来这个问题是怎么解决的呢，我训练自己的数据集也遇到了同样的问题，训练完第一个epoch到validation就卡死了
  
请问这个问题后来你怎么解决的呢？

**Comment by yangshunDragon** [comment]
> 好像是因为影像的原因，把没有标签的剔除掉就可以了
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2022年5月7日(星期六) 下午5:42 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [saic-vul/ritm_interactive_segmentation] It takes long time to train (#5) @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。 The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments. Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。 请问后来这个问题是怎么解决的呢，我训练自己的数据集也遇到了同样的问题，训练完第一个epoch到validation就卡死了 请问这个问题后来你怎么解决的呢？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

没有标签是指有原始图像 images/***.jpg   但是没有对应的掩膜masks/***.png么？

我的情况是使用3D的医学图像切片做的训练数据集，每个原始图像images/***.jpg都有对应的masks/***.png图像，但是mask图像有一定比例是纯黑的（没有目标mask）

**Comment by ty199931** [comment]
就是把那些mask纯黑的删掉就行




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年5月9日(星期一) 上午9:31
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [saic-vul/ritm_interactive_segmentation] It takes long time to train  (#5)





  
好像是因为影像的原因，把没有标签的剔除掉就可以了
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: @.&gt;; 发送时间: 2022年5月7日(星期六) 下午5:42 收件人: @.&gt;; 抄送: @.&gt;; @.&gt;; 主题: Re: [saic-vul/ritm_interactive_segmentation] It takes long time to train (#5) @juntawu 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。 The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments. Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。 请问后来这个问题是怎么解决的呢，我训练自己的数据集也遇到了同样的问题，训练完第一个epoch到validation就卡死了 请问这个问题后来你怎么解决的呢？ — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt;
  
没有标签是指有原始图像 images/.jpg   但是没有对应的掩膜masks/.png么？
 
我的情况是使用3D的医学图像切片做的训练数据集，每个原始图像images/.jpg都有对应的masks/.png图像，但是mask图像有一定比例是纯黑的（没有目标mask）

**Comment by fish-yu-gt** [comment]
> > > > > [@juntawu](https://github.com/juntawu) 请问你训了多久呢，batch_size设置的多大，我这边也遇到了这个问题，我是按照代码中给的训练方式训练的，用了单卡，感觉训练的时间太久了。
> > > > 
> > > > 
> > > > The patch size is set to 32 by default. To save time, you only need to train 55 epochs on COCO_LVIS as the authors did in their experiments.
> > > 
> > > 
> > > Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\last_checkpoint.pth Save checkpoint to experiments\iter_mask\sbd_hrnet18\000_first-try\checkpoints\000.pth 请问我在训练第一个epoch的时候，训练结束后就一直停在这个界面是正常的吗？就是会在这里停滞很久是吗？我也不敢去乱点。
> > 
> > 
> > 请问后来这个问题是怎么解决的呢，我训练自己的数据集也遇到了同样的问题，训练完第一个epoch到validation就卡死了
> 
> 请问这个问题后来你怎么解决的呢？

请问你解决了吗，我的mask没有纯黑的，你是否有其他的解决方案，感谢您的帮助
